### PR TITLE
#168781804 email verification

### PR DESCRIPTION
**Pull request by urbanishimwe**
### What does this PR do?
- user can verify their email before they merge
- documentation of the email verification
#### Description of Task to be completed?
usually, users need to verify their email. to make sure that those emails belong to them
#### How should this be manually tested?
create an account on sendgrid, and generate your api key
user can verify their email through post request by using:
`post:/api/v1/users/verify`
user can verify their email through email sent to them
`get:/api/v1/users/verify/:token`
#### Any background context you want to provide?
NAN
#### What are the relevant pivotal tracker stories?
[story](https://www.pivotaltracker.com/story/show/168781804)
#### Screenshots (if appropriate)
image

#### Questions:
NAN

### CODE REVIEW

**Review by swaibat — COMMENTED** [review]
your coverage is failing by a big percentage

**Review by lemoissonM — APPROVED** [review]

**Review by alexismajya — APPROVED** [review]

**Review by swaibat — APPROVED** [review]

**Review by swaibat — COMMENTED** [review]
test coverage is failing by a big percentage

File: src/controllers/userController.js [hunk 1]
```diff
@@ -43,5 +46,31 @@ const User = {
     }
     return sendResult(res, 400, 'The email and/or password is invalid');
   },
+  async verifyEmail(req, res) {
```
Comment (nakiwuge) [inline src/controllers/userController.js:51]: Add a new line above this line

File: src/utils/stringHelper.js [hunk 2]
```diff
@@ -2,5 +2,11 @@
 function stringHelper() {
   this.signupSuccess = 'Account created successfully';
   this.signupEmailExist = 'This email already exists';
+  // eslint-disable-next-line arrow-parens
+  this.emailBody = token => `
+  <h1> BareFoot Nomad <h1>
+  <h6> number one travel company </h6>
+  <h3> click on the link below to verify your email <h3>
+  <a href ="https://eagle-bn-backend-staging.herokuapp.com/api/v1/users/verify/${token}"> verify email </a>`;
```
Comment (nakiwuge) [inline src/utils/stringHelper.js]: Make the host dynamic by picking the host from the request headers

File: src/routes/userRoutes.js [hunk 3]
```diff
@@ -79,7 +82,62 @@ const app = express.Router();
...
+ *           type: object
+ *           properties:
+ *             msg:
+ *               type: string
+ *       400:
+ *         description: Wrong data sent
+ */
 app.post('/signup', signupValidator, UserMiddle.checkuserExist, userController.signup);
 app.post('/login', UserMiddle.checkloginEntries, userController.login);
+app.post('/verify', userController.requestEmailVerification);
```
Comment (nakiwuge) [inline src/routes/userRoutes.js]: we will not need this route

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@urbanishimwe good job. There is some feedback I would like you to workon.

- Remove the token on signup from the response and just use it in the back to send the email for verification.
- Create a Gmail account for eagle and share the password and email with the team on the channel. 
- The user should only verify their account by accessing the token in the email
- The `/verify/:token` endpoint should be a `put` since you are modifying a field `isVerified` from the database

Also, I have left other comments on the PR kindly check them out.

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**